### ISSUE TITLE
Apply on financial time series data?

### ISSUE THREAD

**Issue by mackancurtaincheeks** [onurpoyraz/chmmMCMC#1]
Would it be possible to implement this with financial time series data?

**Comment by onurpoyraz** [comment]
If the data is discrete, it is possible to implement it with financial time series data.

**Comment by mackancurtaincheeks** [comment]
We are fetching the data from AlphaVantage and YahooFinance so the data consists of a dataframe, but we will use arrays with closing and opening prices. So that should work, right?

**Comment by mackancurtaincheeks** [comment]
Also - is there any kind of readme/guide for the package?

**Comment by onurpoyraz** [comment]
Yes, if you install the package, you can see a quick tutorial and descriptions on the help. I hope it will be sufficient for you. For your information, there will be an update in the following days (it is in the testing phase right now) which will make the package approximately 20 times faster. If you plan to use the package, I recommend you follow the upcoming update.

**Comment by mackancurtaincheeks** [comment]
I will take a look and do hope the tutorial are sufficient to get a working model. Okay, cool! Will you be releasing it within 1-2 weeks? We are doing analysis on the different HMMs we are using in the thesis and it would be cool, if we are able to implement it, to use the new update.

**Comment by onurpoyraz** [comment]
If I do not face with a major problem, I am planning to publish it before the end of this week.

**Comment by onurpoyraz** [comment]
I hope the tutorial is enough, otherwise, you can ask for details.

**Comment by mackancurtaincheeks** [comment]
Is it possible to simulate data from the model?

**Comment by onurpoyraz** [comment]
Using model posterior parameters, it is straightforward to simulate data.

**Comment by mackancurtaincheeks** [comment]
I will give it a try in the morning. I'm new to R so the exampe might be good for somebody a bit more well-versed, but I would need some guidance. Also - what papers did you use for the project? Would be cool to reference.

**Comment by onurpoyraz** [comment]
Inspired by the following papers:
Sherlock-2013-A coupled hidden Markov model for disease interactions
Touloupou2019-Scalable Bayesian Inference for Coupled Hidden Markov and Semi-Markov Models

Additionally, you can check the modeling section of our paper for details:
Poyraz-2021-Modeling MRSA decolonization: Interactions between body sites and the impact of site-specific clearance

**Comment by mackancurtaincheeks** [comment]
Thanks alot! Would it be possible for you to just create some example code with more comments and going through the different features? Doesn't have to be extensive, just a bit more than the example provided with the package. Would really like to use this package in my bachelor's thesis.

**Comment by onurpoyraz** [comment]
I will try to add more details on the documentation of the new version.

**Comment by mackancurtaincheeks** [comment]
That would be awesome! We got a little bit more time to include one more HMM to our thesis.

**Comment by mackancurtaincheeks** [comment]
How is the update going?

**Comment by onurpoyraz** [comment]
Unfortunately, there are still some crucial bugs and I need to contribute to other urgent projects right now. It will take at least 2 weeks to publish a new upgrade.

**Comment by mackancurtaincheeks** [comment]
Damn.. Is it usable?

**Comment by onurpoyraz** [comment]
You can use current version. It is considerably slow but is working.
